From 7f8ca7b863ba1769bc71d384ad34d36caa8c20ee Mon Sep 17 00:00:00 2001
From: Yi Li <yi1.li@intel.com>
Date: Thu, 3 Aug 2023 12:37:34 +0800
Subject: [PATCH 13/18] CryptoPkg: adapt 3.0 change in SslNull.c

Type of input params changed in openssl30.

Signed-off-by: Yi Li <yi1.li@intel.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Xiaoyu Lu <xiaoyu1.lu@intel.com>
Cc: Guomin Jiang <guomin.jiang@intel.com>
Reviewed-by: Jiewen Yao <jiewen.yao@intel.com>
Acked-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Brian J. Johnson <brian.johnson@hpe.com>
Tested-by: Kenneth Lautner <klautner@microsoft.com>
(cherry picked from commit cea8e3b5133546cd0f1ab51b8c3649b13663b51b)
---
 CryptoPkg/Library/OpensslLib/OpensslStub/SslNull.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CryptoPkg/Library/OpensslLib/OpensslStub/SslNull.c b/CryptoPkg/Library/OpensslLib/OpensslStub/SslNull.c
index 49f1405bc0..9f659728fe 100644
--- a/CryptoPkg/Library/OpensslLib/OpensslStub/SslNull.c
+++ b/CryptoPkg/Library/OpensslLib/OpensslStub/SslNull.c
@@ -80,10 +80,10 @@ SSL_CTX_new (
   return NULL;
 }
 
-unsigned long
+uint64_t
 SSL_CTX_set_options (
-  SSL_CTX        *ctx,
-  unsigned long  op
+  SSL_CTX   *ctx,
+  uint64_t  op
   )
 {
   ASSERT (FALSE);
-- 
2.43.0

